### PR TITLE
Add verification for inclusion proof

### PR DESCRIPTION
**Pull request by loosebazooka**
Signed-off-by: Appu Goundan <appu@google.com>

### CODE REVIEW

File: src/main/java/dev/sigstore/rekor/client/RekorVerifier.java [hunk 1]
```diff
@@ -90,6 +92,66 @@ public void verifyEntry(RekorEntry entry)
...
+                    new RekorVerificationException(
+                        "No inclusion proof was found in the rekor entry"));
+
+    var leafHash =
+        Hashing.sha256()
+            .hashBytes(combineBytes(new byte[] {0x00}, Base64.getDecoder().decode(entry.getBody())))
+            .asBytes();
+
+    // see: https://datatracker.ietf.org/doc/rfc9162/ section 2.1.3.2
+    var fn = inclusionProof.getLogIndex();
```
Comment (patflynn) [inline src/main/java/dev/sigstore/rekor/client/RekorVerifier.java:111]: I find the spec algorithm variable names pretty unintuitive
were you tempted to rename them for the java implementation to something more java like?
you could add a comment or something to match them to the spec names

File: src/main/java/dev/sigstore/rekor/client/RekorVerifier.java [hunk 2]
```diff
@@ -90,6 +92,66 @@ public void verifyEntry(RekorEntry entry)
...
+    }
+
+    var calcuatedRootHash = Hex.toHexString(r);
+    if (!calcuatedRootHash.equals(inclusionProof.rootHash())) {
+      throw new RekorVerificationException(
+          "Calculated inclusion proof root hash does not match provided root hash\n"
+              + calcuatedRootHash
+              + "\n"
+              + inclusionProof.rootHash());
+    }
```
Comment (patflynn) [inline src/main/java/dev/sigstore/rekor/client/RekorVerifier.java:141]: should we be verifying the logid using the reko public key?

Comment (loosebazooka) [reply src/main/java/dev/sigstore/rekor/client/RekorVerifier.java:141]: yeah probs. I think I have a TODO somewhere else about that: https://github.com/sigstore/sigstore-java/blob/main/src/main/java/dev/sigstore/rekor/client/RekorVerifier.java#L69

File: src/main/java/dev/sigstore/rekor/client/RekorVerifier.java (hunk 1 above)
Comment (loosebazooka) [reply src/main/java/dev/sigstore/rekor/client/RekorVerifier.java:111]: sgtm, I'll update

**Review by patflynn — APPROVED** [review]

**Pull request merged**